frank-doc-frontend/src/app/frankdoc.types.ts

```diff
@@ -0,0 +1,77 @@
+export interface FrankDoc {
+  metadata: Metadata;
+  groups: Group[];
+  types: TypeElement[];
+  elements: Element[];
+  enums: Enum[];
+}
+
+export interface Element {
+  name: string;
+  fullName: string;
+  abstract?: boolean;
+  description?: string;
+  parent?: string;
+  elementNames: string[];
+  attributes?: Attribute[];
+  children?: Child[];
+  forwards?: Forward[];
+  deprecated?: boolean;
+  parameters?: Forward[];
+  parametersDescription?: string;
+}
+
+export interface Attribute {
+  name: string;
+  mandatory?: boolean;
+  describer?: string;
+  description?: string;
+  type?: TypeEnum;
+  default?: string;
+  deprecated?: boolean;
+  enum?: string;
+}
+
+export enum TypeEnum {
```

Comment:
Waarom hier TypeEnum en niet ChildInterface en ElementInterface?
Maak er anders DataType van?

Reply:
Ik heb meeste types in dit bestand over gekopieerd van de frankflow en bijna niet aangepast, ik op het project board ook nog staan dat ik alle types af moet gaan om te controleren of ze nog wel kloppen.
Sergi had een leuk idee om in de toekomst de types ook te laten genereren voor front-end projecten zoals frankdoc & frankflow